crates/node/src/commands/run.rs

```diff
@@ -119,25 +125,47 @@ impl ExtendedRunCmd {
     }
 }
 
+fn init_da_client(da_layer: DaLayer, da_path: PathBuf) -> Result<Box<dyn DaClient + Send + Sync>> {
+    let da_client: Box<dyn DaClient + Send + Sync> = match da_layer {
+        DaLayer::Celestia => {
+            let celestia_conf = CelestiaConfig::try_from(&da_path)?;
+            Box::new(CelestiaClient::try_from(celestia_conf).map_err(|e| sc_cli::Error::Input(e.to_string()))?)
+        }
+        DaLayer::Ethereum => {
+            let ethereum_conf = EthereumConfig::try_from(&da_path)?;
+            Box::new(EthereumClient::try_from(ethereum_conf)?)
+        }
+        DaLayer::Avail => {
+            let avail_conf = AvailConfig::try_from(&da_path)?;
+            Box::new(AvailClient::try_from(avail_conf).map_err(|e| sc_cli::Error::Input(e.to_string()))?)
+        }
+    };
+
+    Ok(da_client)
+}
+
 pub fn run_node(mut cli: Cli) -> Result<()> {
     if cli.run.base.shared_params.dev {
         override_dev_environment(&mut cli.run);
     }
     let runner = cli.create_runner(&cli.run.base)?;
 
-    let da_config: Option<(DaLayer, PathBuf)> = match cli.run.da_layer {
+    let (da_config, _da_client) = match cli.run.da_layer {
```

Comment:
`_da_client` should be used and passed as an argument to `service::new_full`.
The content of `init_da_client` is duplicated in `service.rs` where it should be removed.